### DISCUSSION TITLE
Why the 'L2 miss' sums three events?

### DISCUSSION THREAD

**Discussion by glarer** [opcm/pcm#312] [Q&A] [answered]
### L2MIS in 'cpucounter.h' using three events to count.  

- 'L3Miss', is it the 'Event num: 2E, Umask: 41' ? 
- 'L3UnsharedHit', I'm sorry I can't find the event to match it.
- 'L2HitM', I also can't find a event to match this.

My question is why not use the event 'L2_RQSTS.MISS' (Event num:24 Umask 3F)? The description 'All requests that missed L2.'  is more likely the 'L2 miss'.
Thanks a lot.

**Comment by opcm** [comment, marked as answer]
L3 cache does not have events directly comparable to L2_RQSTS.MISS on many architectures, therefore to provide stats that can be compared across L2 and L3 other events were used.

**Comment by glarer** [comment]
Skylake 😂😂




------------------ Original ------------------
From: opcm ***@***.***&gt;
Date: Wed,Jul 28,2021 3:59 PM
To: opcm/pcm ***@***.***&gt;
Cc: BiYX ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [opcm/pcm] Why the 'L2 miss' sums three events? (#312)





 
on which CPU architecture did you consider to use L2_RQSTS.MISS ?
